src/Compilers/Extension/SetGlobalGlobalPropertiesForCPS.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Roslyn.Compilers.Extension
+{
+    [ExportBuildGlobalPropertiesProvider]
+    [AppliesTo("(" + ProjectCapabilities.CSharp + " | " + ProjectCapabilities.VB + ")" + " & " + ProjectCapabilities.LanguageService)]
+    public class SetGlobalGlobalPropertiesForCPS : StaticGlobalPropertiesProviderBase
+    {
+        [ImportingConstructor]
+        public SetGlobalGlobalPropertiesForCPS(IProjectService projectService)
+            : base(projectService.Services)
+        {
+        }
+
+        public override Task<IImmutableDictionary<string, string>> GetGlobalPropertiesAsync(CancellationToken cancellationToken)
+        {
+            return CompilerPackage.RoslynHive != null
```

Comment:
Why do we think the compiler package loads and set the static prior to this running? Can whatever mechanism it is be commented so we understand why?

Reply:
Today, project load (`SolutionExists`) in the shell always runs before CPS. CPS hasn't even created the project system until after the shell is done loading things.  I will add a comment so we can know to update this if this logic ever changes.  Truthfully should just change both project systems so that per-hive properties like this are easy to do correctly.